### PR TITLE
fix(appStart): Handle undefined AppRegistryIntegration.onRunApplication gracefully

### PR DESCRIPTION
**Pull request by antonis**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description

Handle undefined AppRegistryIntegration.onRunApplication edge case gracefully.

## :bulb: Motivation and Context

While working on https://github.com/getsentry/sentry-react-native/pull/4604, I've noticed that after merging https://github.com/getsentry/sentry-react-native/pull/4641 from `main` the `e2e-tests/RnDiffApp` crashed on startup when the root app component is wrapped with  `Sentry.wrap`.

## :green_heart: How did you test it?
Manual, CI

## :pencil: Checklist

- [ ] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] I updated the docs if needed.
- [x] I updated the wizard if needed.
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

#skip-changelog

### CODE REVIEW

File: packages/core/src/js/tracing/reactnativeprofiler.tsx [hunk 1]
```diff
@@ -47,9 +47,17 @@ export class ReactNativeProfiler extends Profiler {
       return;
     }
 
+    const appRegistryIntegration = getAppRegistryIntegration(client);
+
+    if (!appRegistryIntegration || typeof appRegistryIntegration.onRunApplication !== 'function') {
+      // eslint-disable-next-line no-console
+      __DEV__ && console.warn('AppRegistryIntegration.onRunApplication not found or invalid.');
```
Comment (antonis) [inline packages/core/src/js/tracing/reactnativeprofiler.tsx:54]: Covering this edge case with a test proved difficult since the mocking affected other test case. I'll be happy to revisit if this is a scenario we should cover.

<details><summary>example</summary>
<p>

```
test('handles graceful missing onRunApplication', async () => {
  const invalidMockedGetAppRegistryIntegration = jest.spyOn(AppRegistry, 'getAppRegistryIntegration').mockReturnValue({
    name: 'appRegistryIntegrationMocked',
    onRunApplication: undefined,
  });

  const testClient = new TestClient(getDefaultTestClientOptions());

  reactNavigationIntegration().afterAllSetup(testClient);

  expect(invalidMockedGetAppRegistryIntegration).not.toHaveBeenCalled();
});
```

</p>
</details>

Comment (krystofwoldrich) [inline packages/core/src/js/tracing/reactnativeprofiler.tsx:54]: Since this is expected to run when the client exists, we can use the logger.

File: packages/core/src/js/tracing/reactnativeprofiler.tsx [hunk 2]
```diff
@@ -47,9 +47,17 @@ export class ReactNativeProfiler extends Profiler {
...
+      return;
```
Comment (krystofwoldrich) [inline packages/core/src/js/tracing/reactnativeprofiler.tsx:55]: We should not `return`, because when tracing is disabled, we still won't the rest to execute.

File: packages/core/src/js/tracing/reactnativeprofiler.tsx [hunk 3]
```diff
@@ -47,9 +47,17 @@ export class ReactNativeProfiler extends Profiler {
...
+    }
+
     client.addIntegration && client.addIntegration(createIntegration(this.name));
 
-    getAppRegistryIntegration(client).onRunApplication(ReactNativeProfilerGlobalState.onRunApplicationHook);
+    appRegistryIntegration.onRunApplication(ReactNativeProfilerGlobalState.onRunApplicationHook);
```
Comment (krystofwoldrich) [inline packages/core/src/js/tracing/reactnativeprofiler.tsx:60]: If the `return` will be removed.
````suggestion    appRegistryIntegration?.onRunApplication(ReactNativeProfilerGlobalState.onRunApplicationHook);
````

File: packages/core/src/js/tracing/reactnativeprofiler.tsx (hunk 1 above)
Comment (antonis) [reply packages/core/src/js/tracing/reactnativeprofiler.tsx]: Thank you for the hint Krystof 🙇 
Added a test case in [`wrap.test.tsx`](https://github.com/getsentry/sentry-react-native/pull/4743/files#diff-e7bfab793e965f5809e6743967f86e4f8ec0bee003111b92525e6b1c9f65c638)

**Review by krystofwoldrich — APPROVED** [review]
LGTM 🚀 Thank you for the catch!
